### DISCUSSION TITLE
Fetching  Vercel Edge config on the client side

### DISCUSSION THREAD

**Discussion by ashishkankal** [vercel/community#1542] [Help] [answered]
Is it possible to fetch the Vercel Edge config on the client side? Tried the below approach. But doesn't seem to be working
```
import { get } from "@vercel/edge-config";
...
const Componnent = ()=> {
     ...
      useEffect(() => {
           async function loadConfig() {
           const configVar = await get("configVar")
     }
    loadConfig();
  });
   ...
}
```
**Error**
<img width="937" alt="image" src="https://user-images.githubusercontent.com/7954909/218982678-bb6a4f63-71f0-43c5-91f7-36bfae6fee38.png">

**Comment by dferber90** [comment, marked as answer]
It is possible but it has a significant tradeoffs:
- you will need to publish your connection string to the client
  - **this means anyone will be able to read your whole edge config**
- it will of course need go over the network, so it will not be instant

When you do `import { get } from "@vercel/edge-config"` it is actually doing the following behind the scenes:

```ts
import { createClient } from "@vercel/edge-config"
const { get } = createClient(process.env.EDGE_CONFIG)
```

The EDGE_CONFIG variable is not available on the client. That's why you're seeing the "No connection string provided".

> ⚠️  You can go to the project settings and rename the EDGE_CONFIG environment variable to `NEXT_PUBLIC_EDGE_CONFIG`. This will publish the connection string to the client. Be careful that this will allow anyone to read the whole content of your Edge Config!

You can then access the Edge Config like so

```ts
import { createClient } from "@vercel/edge-config"
const { get } = createClient(process.env.NEXT_PUBLIC_EDGE_CONFIG)
await get("your_key")
```

**An alternative approach**: If you want to avoid publishing your Connection String to the client you can instead create an [Edge API Route](https://nextjs.org/docs/api-routes/edge-api-routes) which reads the desired keys from your Edge Config and responds with them. It will have almost identical performance to publishing the Connection String and reading the Edge Config directly. This approach also allows you to add custom cache headers.
